src/utils/search_conda.ts

```diff
@@ -0,0 +1,46 @@
+import { toast } from "react-toastify";
+
+export type CondaApiPackage = {
+  name: string;
+  latestVersion: string;
+  url: string;
+  owner: string;
+};
+
+async function fetchCondaPackages(query: string): Promise<CondaApiPackage[]> {
+  const searchUrl = `https://api.anaconda.org/search?name=${encodeURIComponent(query)}&channel=conda-forge`;
+
+  try {
+    const response = await fetch(searchUrl);
+
+    if (!response.ok) {
+      throw new Error(`HTTP error! status: ${response.status}`);
+    }
+
+    return await response.json();
+  } catch (error) {
+    const errorMessage =
+      error instanceof Error ? error.message : "An unknown error occurred.";
+    toast.error(`Error fetching data from Anaconda.org: ${errorMessage}`);
```

Comment:
have you tested this toast? can you send a screenshot? it looks like it is running on the server

Reply:
You are right, toast does not work server side, I tried propagating the error up to the client side page but the error does not propagate with debounce: https://github.com/lodash/lodash/issues/4449. I looked into the suggested solution and tried implementing that but the error would not propagate: https://github.com/lodash/lodash/issues/4815#issuecomment-1006573679